deployment/docker/Makefile

```diff
@@ -12,7 +13,11 @@ package-builder:
 	(docker image ls --format '{{.Repository}}:{{.Tag}}' | grep -q '$(BUILDER_IMAGE)$$') \
 		|| docker build -t $(BUILDER_IMAGE) deployment/docker/builder
 
-app-via-docker: package-certs package-builder
+package-rootless:
+	(docker image ls --format '{{.Repository}}:{{.Tag}}' | grep -q '$(ROOTLESS_IMAGE)$$') \
+		|| docker build -t $(ROOTLESS_IMAGE) deployment/docker/rootless
```

Comment:
I think `certs_image` and `rootless_image` may be collapsed together

Reply:
How about making some basic shared image, e.g.:

##### `deployment/docker/shared-base/Dockerfile`
```Dockerfile
# See https://medium.com/on-docker/use-multi-stage-builds-to-inject-ca-certs-ad1e8f01de1b
FROM alpine:3.10 AS prepare_stage
RUN apk --update add ca-certificates
RUN adduser -u 1000 -S -h /home/victoriametrics victoriametrics
USER 1000

FROM scratch
COPY --from=prepare_stage --chown=0:0 /etc/passwd /etc/passwd
COPY --from=prepare_stage --chown=0:0 /etc/group /etc/group
USER 1000
COPY --from=prepare_stage --chown=1000 /home/victoriametrics /home/victoriametrics
COPY --from=prepare_stage /etc/ssl/certs/ca-certificates.crt /etc/ssl/certs/ca-certificates.crt
```
This will create clear base image, which fs-layer can be reused over all `vm*` images, e.g.:

##### `app/vminsert/deployment/Dockerfile`
```Dockerfile
ARG base_image
FROM $base_image

ARG src_binary
COPY $src_binary ./vminsert-prod

EXPOSE 8480
ENTRYPOINT ["/vminsert-prod"]
```

In this case you are not only sharing same base fs-layer across all images but also make Dockerfile's smaller and more clear to understand

Reply:
Agreed with this idea. Let's do it!